storage/innobase/include/trx0trx.h

```diff
@@ -1031,7 +1039,10 @@ struct trx_t{
 	os_event_t	wsrep_event;	/* event waited for in srv_conc_slot */
 #endif /* WITH_WSREP */
 
-	bool vtq_notify_on_commit;	/*!< Notify VTQ for System Versioned update */
+	/* System Versioning */
+	bool		vtq_notify_on_commit;
+					/*!< Notify VTQ for System Versioned update */
+	vtq_query_t	vtq_query;
```

Comment:
Memory is cheap, time is precious. Yes, I decided that it's worth (since I made it). Whether *really* or not is not possible to answer without real statistics data.